# --- Trajectory planning thresholds ---
max_cruise_vel: 0.4
max_cruise_acc: 0.5
sp_rate: 50.0
traj_rate: 20.0
transform_filter_rate: 20.0

# --- Cooperative robots ---
cobot_1: 22
cobot_2: 26
corr_z: true

# thresholds for "arrived state"
pos_threshold: 0.1 #meters
yaw_threshold: 0.2 #degree
height_threshold: 0.15 #meters
max_height: 3.0
drone_height: 0.21
marker_time_dt: 0.5 #max difference between marker timestamp and time

# TOF params DA MISURARE
critical_distance: 0.30 # m
angle_tof_0: 1.0472  #rad
angle_tof_1: 2.0944  #rad
angle_tof_2: 2.6180  #rad
angle_tof_3: 3.6652  #rad
angle_tof_4: 4.1888  #rad
angle_tof_5: 5.2360  #rad
angle_tof_6: 5.7596  #rad
angle_tof_7: 0.5236  #rad

# --- Mavros ---
setpoint_topic: /mavros/setpoint_raw/local
pose_topic: /mavros/vision_pose/pose
mavros_state_topic: /mavros/state

# --- Aruco Detection ---
markers_topic_name: /aruco_marker_publisher/markers
markers_list_topic_name: /aruco_marker_publisher/markers_list

# --- Visual Servoing ---
rate_servoing: 30
Kp_vs: 10.0    #Proportional gain 45
Kd_vs: 2.0    #Derivative gain 10
norm_threshold: 0.50 #norm threshold for the PD out

# --- Land on Marker ---
theta_max: 0.2  #max cone angle
vel_max: 0.5    #max landing speed
min_height: 1.0 #min height to land without visual servoing

# --- Markers positions ---
#Posizione riferita al centro del marker
marker_1_id: 1
marker_1_x: 0.5
marker_1_y: 3.5
marker_1_z: 0.0

marker_2_id: 2
marker_2_x: 18.5
marker_2_y: 2.5
marker_2_z: 0.0

marker_3_id: 3
marker_3_x: 7.5
marker_3_y: 0.5
marker_3_z: 0.0

marker_4_id: 4
marker_4_x: 1.5
marker_4_y: 7.5
marker_4_z: 0.0

marker_5_id: 5
marker_5_x: 4.5
marker_5_y: 5.5
marker_5_z: 0.0

marker_6_id: 6
marker_6_x: 17.5
marker_6_y: 7.5
marker_6_z: 0.0

marker_7_id: 7
marker_7_x: 15.0
marker_7_y: 1.0
marker_7_z: 0.7

marker_8_id: 8
marker_8_x: 9.0
marker_8_y: 8.0
marker_8_z: 0.0

marker_9_id: 9
marker_9_x: 12.5
marker_9_y: 9.5
marker_9_z: 0.0

marker_10_id: 10
marker_10_x: 4.5
marker_10_y: 1.5
marker_10_z: 2.0
